controllers/observability_controller.go

```diff
@@ -225,21 +224,21 @@ func (r *ObservabilityReconciler) InitializeOperand(mgr ctrl.Manager) error {
 		return err
 	}
 
-	storageClassExists := false
-	storageClass := &storage.StorageClass{}
-	selector := client.ObjectKey{
-		Name: storageClassGp2,
+	runningOnCloud := false
+	cluster := client.ObjectKey{
+		Name: "cluster",
 	}
+	infra := &infrastructure.Infrastructure{}
+	err = r.Get(context.Background(), cluster, infra)
 
-	err = apiReader.Get(context.Background(), selector, storageClass)
-	if err == nil {
-		// no error means that the storage class was found
-		storageClassExists = true
-		r.Log.Info(fmt.Sprintf("found storage class %s on the cluster", storageClassGp2))
+	// Check the infrastructure, If its Libvirt that means it is running on crc so it will run without storage.
+	if err == nil && string(infra.Status.PlatformStatus.Type) != string(infrastructure.LibvirtPlatformType) && string(infra.Status.PlatformStatus.Type) != string(infrastructure.NonePlatformType) {
+		runningOnCloud = true
 	}
```

Comment:
Instead of having two separate if statements, maybe we could use an if/else here:
```suggestion
	instance := observabilityInstanceWithStorage(namespace)
	// Check the infrastructure. If it's Libvirt, it is running on crc, so it will run without storage.
	if err == nil && string(infra.Status.PlatformStatus.Type) != string(infrastructure.LibvirtPlatformType) && string(infra.Status.PlatformStatus.Type) != string(infrastructure.NonePlatformType) {
		runningOnCloud = true
	} else {
		r.Log.Info("Running without Storage.")
		instance = observabilityInstanceWithoutStorage(namespace)
	}
```

Reply:
Makes sense ill make and test those changes and push them up.